### DISCUSSION TITLE
[Balance] The Economy

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#8739] [Unstable feedback]
Here we try to talk about what possibly breaks the economy of Barotrauma

# Free Stuff:

**Respawn shuttles print money**
Respawn shuttles are fun, respawn shuttles keep MP servers alive, but they spawn with a ton of fresh items. 
Expect players to grab every single item off the Hemulen, including diving suits and head to the deconstructor for free titanium alloy.
![image](https://user-images.githubusercontent.com/73229309/161979137-d5e5e5c7-a2c5-4107-bcef-5e22a5acb09d.png)

Reduce the amount of lootables (oxygen masks aren't needed).
Instead of spawning regular/abyss suits, use _Disposable Diving suits_
Doesn't deconstruct into materials, can't be refilled, comes with its own oxygen source, resists all depth. 

**Security is loaded**
Every time a player joins, they get their starter kit's worth of items.
Security spawns with a ton of valuable gear that can be sold. Sec 1 is the most loaded of all, clocking at 1600 Mks' worth.
![image](https://user-images.githubusercontent.com/73229309/161981873-bc6b14b4-04e9-4705-ac74-821d2fd9028c.png)

Rebalanced security loadouts:
Sec 1: Clothes, SMG+Magazine, Knife, Bandage 
Sec 2: Clothes, Shotgun+6 shells, Knife, Bandage
Sec 3: Clothes, Stun baton, Handcuffs, Body armor, Helmet, Bandage 

**Headsets and clothes**
Everyone spawns with clothes and headsets. These can be sold for money. These sell for about 400mk alone, can be found on wreck corpses, and give you endless quantities of batteries,lithium, plastic...
Basic clothes should sell for 0, headsets not use batteries, not deconstructible/craftable) (you'll always find more on dead bodies or a shop if you lose yours)

**Plants**
Nowhere as broken as it used to be, there are still no growth limitations to plants. 
Not uncommon to see every pixel of the engine room being spammed with tobacco for printing money.
Somehow find a way to curb the pot spam? Fertilizer? Refill the watering can in ballasts?

-----

# Inventory management

**Recursive inventory**
You can easily get absurd amounts of inventory slots in crates for siphoning every single item in wrecks.
It's not just convenience, it allows a single diver to take everything in one trip.
See https://github.com/Regalis11/Barotrauma/discussions/6593

**Akimbo duffles**
This is what peak wreck looting looks like:
![image](https://user-images.githubusercontent.com/73229309/161986984-62186d01-4593-49a1-abb8-2fb0bb0b5ce2.png)

Duffels need to be worse than crates in every aspect.
Reduce duffel bag to 12 slots (10 slots+armor+head) make it 2-handed. Toolbelt drops on floor

**Corpses**
Wreck corpses are like regular player corpses: they have 10 inventory slots, clothes, headsets, can wear toolbelts and diving suits.
As seen above you can carry a corpse as a looting backpack on top of your crate/scooter/duffle
[Use actual corpses](https://steamcommunity.com/sharedfiles/filedetails/?id=2736549126), reduce the corpse inventory to 3 slots, no belt slot, no suit slot

**The floor is free real estate**
You can have infinite inventory space by dumping items/crates on the floor. It doesn't matter if in theory your sub has limited ammunition/crate storage. It is very common to see dozens of backup ammunition boxes littered next to loaders:
![image](https://user-images.githubusercontent.com/73229309/161991738-d1eac747-665b-430a-9968-41af2b52a073.png)

Ammunition boxes should lose condition when tossed around by impact. Crates should ragdoll and damage players when sent flying?

-----

# Lootable content

**Ruins**
Ruins are absolutely LOADED with expensive items. Once it becomes easy to kill guardians (masterwork weapons), they are an insane source of income. Xenologist medic talents takes that to 11. This is one room:
![image](https://user-images.githubusercontent.com/73229309/161993287-25043646-d7a3-4ea6-ae28-68895942503f.png)
Late game biomes (Plateau, Wastes and Great Sea) have a 100% chance to spawn ruins as well.

Make ruins much rarer, more dangerous to match the reward.

**Wrecks**
Only 5 wreck types with 20% chance of spawning each, 4 of them being medium-large.
Berilia Wreck worth of loot (5 large cabinets 3 medium)
![image](https://user-images.githubusercontent.com/73229309/161998435-5621ed69-aec5-45e6-9250-da00d4a3a87d.png)

Also spawns with lots of pristine, free diving suits which you can equip on corpses and deconstruct:
![image](https://user-images.githubusercontent.com/73229309/162285111-50730177-1173-436f-8d8d-e4d2b3899023.png)

Need a way to set spawn chance for invidiual wrecks (like outpost modules), dilute the wreck pool so smaller wrecks spawn more often

**Mining**
Mining missions aren't balanced, the difference in rewards is huge
Quartz (20 silicon) is worthless
Pyromorphite (20 chlorine) isn't great
Titanite gives you 18 precious titanium 
Thorianite/Uranium mission means fuel for days (16 Uranium, 12 Thorium)

Change the minerals or rebalance the amount and rewards.

**Beacons**
Maybe you shouldn't be able to steal several missions' worth of maintenance equipment out of a beacon; beacons are treated like mini-wrecks.
![image](https://user-images.githubusercontent.com/73229309/161990047-d91462f0-4ab4-4a74-8881-08bcee7cccb4.png)

Reduce the amount of loot in beacons or lose reputation for stealing

---

# Other stuff

**Fuel hoarding**
It's easy to hoard a campaign's worth of fuel very early in the game. One good mining mission is enough.
Supply and demand of fuel as a resource should kept tense through the campaign.

Make Uranium/Thorianite minerals generate radiation, make fuel rods outside of reactor decay after a delay. 
Fuel will be renewed much more regularly.

**Abusing Cargo missions**
The stuff inside cargo crates is much more valuable than the reward you get.
You can fabricate it into items that can be sold instead.
![image](https://user-images.githubusercontent.com/73229309/162012266-9f8acbeb-db1d-48c7-90c6-b11a9285c140.png)
The items themselves aren't really 

Replace those items by Cargo-specific items or crates. Some water-sensitive.
Hurt reputation in case of mission failure?

**Deconstructing** 
Deconstructing. The fact that when you deconstruct something, you gain 100% of the items needed to craft it.
Removing that would solve a lot of problems.

See this code ? It should be applied to every. single. item
![image](https://user-images.githubusercontent.com/73229309/163357421-450396ab-e5c7-41c3-b406-8a84780dbeac.png)

(the "amount" would be N-1, where N is the amount of different items needed to craft something).
In the case of a scooter, it would yield either 1 alu, or either 1 plastic
The welding tool would look like that:
![image](https://user-images.githubusercontent.com/73229309/163357586-9187fc23-4e99-4921-bf12-371fc76ec57e.png)

Reducing deconstruction yields prevents deconstructing the entire inventory of tools (welders, oxygen bottles) to make them all masterworks without loss, reduces the yield of scavenged/stolen items (from wrecks/outposts and even respawn shuttles) and makes scrap, mining and purchasing supplies the more reliable way of getting resources.

Also prevents farming skills from repeating fabricating and deconstructing

**Outposts**
Stealing is very easy and rewarding. Most of the time you won't get punished by reputation loss as you keep moving forward, never going back. The outposts are packed with supplies
![image](https://user-images.githubusercontent.com/73229309/163942484-d7fc48b4-524f-41af-8f96-1afdd4375ecb.png)

Lossless deconstructing is once again used to make money and materials off stolen supplies:
![image](https://user-images.githubusercontent.com/73229309/163942686-d3ecb3f7-9654-4e05-8eaf-f4c07551a7f6.png)

**Shops**

**Submarine costs**

**Comment by JeroonBosch** [comment]
Thanks for the feedback, many useful comments in here.

I think a lot of the issues here relate to how profitable selling is at 80% of base price value.

The inventory management, while silly in it's current implementation, at least reduces the amount of time needed to loot a place. Since looting would otherwise takes a lot of time, I personally think we should adjust the amount of spawned loot before we address inventory space.

I think some short-term improvements to look into first are:
- Decrease selling price
- Less gear on respawn shuttles
- Less gear on beacon stations
- Less value of headsets in particular (in addition for the overall decreased selling price)

The other points will either need some discussion/thought or a large amount of value changes.

**Comment by mLuby** [comment]
As always, overall great points and presentation.

> [For respawn shuttles] reduce the amount of lootables (oxygen masks aren't needed).
> Instead of spawning regular/abyss suits, use **_Disposable Diving suits_**
> Doesn't deconstruct into materials, can't be refilled, comes with its own oxygen source, resists all depth.

> Every time a player joins, they get their starter kit's worth of items.

Respawning players should get assistant's clothes and a headset, that's it. Of course if it's the player's first spawn in this game they can get their full starter kit.

Disposable Diving Suits are a great idea! They make tons of sense for a world where breaches and flooding are a common occurrence. Like [inflatable liferaft barrels](https://en.wikipedia.org/wiki/Lifeboat_(shipboard)#Liferaft_versus_lifeboat) on modern ships.

> Duffels need to be worse than crates in every aspect.

Or make Duffle bags deteriorate quickly in water till they disappear, spilling their contents. (Also please make them fit in crate shelves.)

> Use actual corpses, reduce the corpse inventory to 3 slots, no belt slot, no suit slot.

Just make carrying a corpse require both hands like crates or toolbelts. If you really want to keep the "drag someone on a scooter" gameplay, make the passenger do the grabbing, not the driver as it currently works. And if you're concerned about retrieving divers quickly, maybe Medics could make a [Medical Stretcher](https://en.wikipedia.org/wiki/Stretcher) Scooter that can carry a body as cargo.

> Ammunition boxes should lose condition when tossed around by impact. Crates should ragdoll and damage players when sent flying?

I'd go so far as to say that _every_ object moving through air should have a chance to damage/stun characters (perhaps based on its mass/size). A more hilariously evil option is to give tossed crates a chance to spill their contents on impact.

> Supply and demand of fuel as a resource should kept tense through the campaign.

Pro players may forget that getting soft-locked by lack of fuel is both super frustrating and surprisingly easy. I'd prefer to penalize the hoarding itself: if you keep too many fuel rods in the same place, they might spontaneously go supercritical, creating radiation and fire until they're exhausted or separated by someone's brave sacrifice.

> The stuff inside cargo crates is much more valuable than the reward you get.

Replace the cargo crates with the [Secure Metal Crates](https://barotraumagame.com/wiki/Crates) already the game.

**Reply by ghost** [reply]
I think the no gear on respawn idea should only apply to campaign.

**Comment by cshingle** [comment]
I personally feel that making supplies decay will just add frustration to the game for most players and not increase the fun. I suggest addressing these issues in other ways such lowering selling prices, adjusting how sellers buy.

I think a great way of reducing shop abuse is to limit the number of any specific item they are willing to buy. A good way to implement this is to have the shop want to stock all items up to level, but not above it. For example a mining station may want to buy 5 shotguns since they don't have any for sale, but won't buy steel because they have already reached their sale limit.

Storing stuff on the floor is less of an issue in my mind because most servers I am on/run there is a push to clean up the sub to prevent loosing items via breaches and slowing down clients with potato PCs. There is also the issue of stuff getting suck in the floor.
I think making larger items cause damage to people and/or the sub could add to the fun.

> Replace the cargo crates with the [Secure Metal Crates](https://barotraumagame.com/wiki/Crates) already the game.

I think this is a good idea for most missions. I do like the idea of having missions able to have you transport volatile materials in non-waterproof creates and crazy stuff like that.

**Comment by LePiromano** [comment]
so true

**Comment by ubertpendragon** [comment]
Added a part about deconstructing from Renko

**Comment by Renko-6662** [comment]
Thanks for adding my deconstructing idea.

For plants abuse, I feel the best way would be to limit to amount of planters. Kinda like the mechanic's portable pump, which are limited to 2, the planters could be limited to Hull Volume / X

I feel the game also need something big : 

### Price overhaul

Some prices don't makes sense at all. Why is tin, which is useless, more expensive than carbon (10 vs 11 mk) ? Why is uranium (50mk) be almost the same price as plastic (45 mk) ? Why is thorium (125 mk) almost the same price as a common alien blood (120 mk) ?

But more importantly, this prices can also lead to weird things. Take artifacts for example :

1 Artifact = 1000 mk
1 Artifact deconstructed = 3 alien materials = 600 mk

Deconstructing an artifact can be dangerous. Why is it paying less than just selling the artifact ? But wait, there's more !
There's a medic talent that grants you 50% chance to double the output when deconstructing an artifact. So let's say a statistically 50% flat increase. So we got :

1 Artifact deconstructed with medic talent = 4.5 alien materials = 900 mk... still less than selling the artifact !

**My suggestion, although it might be complicated to implement :**

Make prices dynamic. We'll take the welding tool as an example, which is crafted with a steel and a plastic.
Right now (in unstable), prices are : Welding tool : 75 mk, Steel : 20 mk, Plastic : 45 mk
So right now the welding tool cost 65 mk to craft, 75 mk to buy, and you can't changes prices without affecting the whole economy and having to balance everything. Instead, I would suggest this changes :

- Changes prices from static prices to dynamic ones.. The welding tool new price would be :
[Price of a Steel] + [Price of a Plastic] + "Crafting fee"

- Use your game analytics to analyse what are the most sold / the most profitable items, or the less used ones. Then, either change the recipe of problematic items, or change one of their components prices, and the item price as well as the whole economy, will be affected by the change.

AND / OR
- Use your game analytics to watch what are players selling in this campaign. For example, each 10 Steel sold, the price of steel could come down by one mark, or for each 10 Uranium bought, the price or uranium could come up by one mark. (That could allow you to also implement dynamic events, like a shortage or a surplus, and you could even tie this system with factions, each faction could have its own materials prices). And for people making crafted items before selling them, it could affect the price of its components. Like welding tools sold / bought would still affect the steel and plastic prices.

**Reply by mLuby** [reply]
I like the deconstructing idea a lot. Worth mentioning that non-fully-reversible crafting limits experimentation so it's a slight detriment to newer players.

> Deconstructing an artifact can be dangerous. Why is it paying less than just selling the artifact

Worth more than the sum of its parts? Sounds correct to me. Especially since artifacts, which have powers, can't be crafted from alien materials, which don't have powers.

> Make prices dynamic. Use your game analytics to analyse what are the most sold / the most profitable items

Sort of a shadow MMO economy—no direct player interaction but prices change based on other players' interactions with their worlds.
1. I don't think they capture analytics data at such a granular level. Not that it couldn't happen.
2. They'd probably have to throw away data from modded servers so it might not be as much usable data as you'd expect, and might skew toward newer players who aren't using mods yet.
3. Would only non-modded prices be dynamic? It could create a multiplier (e.g. 1.15*base price) that changes even modded prices. Would dynamic prices be something a setting or mod can disable?
4. It still needs some human tuning. They can't let bandages or fuel rods or welding fuel become too expensive just because everyone is always buying these essentials—in economic terms they're [price inelastic](https://en.wikipedia.org/wiki/Price_elasticity_of_demand#Determinants).

It'd be interesting if in such a shadow MMO economy, stations _with the same name_ were linked. So if I dump 20 alien artifacts into Undermoist's local economy, that's reflected for other players who visit Undermoist (which, being randomly generated, is likely in a different location or even biome for them).

**Comment by Ek-Videogames** [comment]
I'm disappointed that nobody's talking about nonsensical shop prices & deconstruction loop exploits.

Also maybe a mission-economy / jovian-radiation / mining resources rebalance & fun-balance.

**Reply by Ek-Videogames** [reply]
- when I said fixed shop prices I meant solving the arbitrary 300% markup on things like coilgun ammo and other items that are way cheaper than the items they deconstruct into, **not removing the deconstructor & item selling from the game**

**Reply by cshingle** [reply]
I see the unstable changes that "breaks" the deconstruction loop "exploits" as a tragedy. There are lots of potential solutions to it such as adjust construction times, or cap skill gain per minute/round, but instead it was chosen to remove a feature that allows new players to learn the game. Now new players will not be welcome in some roles because they will "waste" resources.

I don't really see people buying stuff, deconstructing it and selling it back. That requires a undock, redock loop that I just don't see on the servers I play on. If people are going to go though the effort to do that, then they are just as likely to spend less time to go edit the save file. The goal should be to maximize the fun, not chase stupid "exploits".

I think the shop prices and fabricator recipes need a pass for their own sake. There is a lot that doesn't make sense and should have the base cost or recipe changed. Why does my harpoon gun carry 6 railgun shells worth of steel, and my revolver 1.5 coil gun boxes of lead?
